tests/Integration/Behaviour/Features/Scenario/Feature/feature_management.feature

```diff
@@ -8,22 +8,121 @@ Feature: Product feature management
-    Then I should get an error that feature name is invalid.
+      | name[en-US] | My feature en |
+      | name[fr-FR] | My feature fr |
+    Then product feature feature1 should have following details:
+      | name[en-US]      | My feature en |
+      | name[fr-FR]      | My feature fr |
+      | associated shops | shop1         |
+
+  Scenario: Update product feature
+    Given I create product feature "feature2" with specified properties:
+      | name[en-US] | My feature en |
+      | name[fr-FR] | My feature fr |
+    And product feature feature2 should have following details:
+      | name[en-US]      | My feature en |
+      | name[fr-FR]      | My feature fr |
+      | associated shops | shop1         |
+    When I update product feature feature2 with following details:
+      | name[en-US] | My feature en updated1 |
+      | name[fr-FR] | My feature fr updated1 |
+    Then product feature feature2 should have following details:
+      | name[en-US] | My feature en updated1 |
+      | name[fr-FR] | My feature fr updated1 |
+
+  Scenario: Creating and updating feature with empty shop association should assign the feature to current shop context by default
+    When I create product feature "feature3" with specified properties:
+      | name[en-US]      | My feature en |
+      | name[fr-FR]      | My feature fr |
+      | associated shops |               |
+    Then product feature feature3 should have following details:
+      | name[en-US]      | My feature en |
+      | name[fr-FR]      | My feature fr |
+      | associated shops | shop1         |
+    When I update product feature feature3 with following details:
+      | associated shops |  |
+    Then product feature feature3 should have following details:
+      | name[en-US]      | My feature en |
+      | name[fr-FR]      | My feature fr |
+      | associated shops | shop1         |
+
+  Scenario: Creating and updating product feature with empty name in default language should not be allowed
+    When I create product feature "feature4" with specified properties:
+      | name[en-US] |               |
+      | name[fr-FR] | My feature fr |
+    Then I should get an error that feature name is invalid
+    Given I create product feature "feature4" with specified properties:
+      | name[en-US] | My feature en |
+      | name[fr-FR] | My feature fr |
+    And product feature feature4 should have following details:
+      | name[en-US] | My feature en |
+      | name[fr-FR] | My feature fr |
+    When I update product feature feature4 with following details:
+      | name[en-US] |               |
+      | name[fr-FR] | My feature fr |
+    Then I should get an error that feature name is invalid
+    And product feature feature4 should have following details:
+      | name[en-US] | My feature en |
+      | name[fr-FR] | My feature fr |
+
+  Scenario: Creating product feature with empty name in default language should not be allowed
```

Comment:
Isn't it redundant with the previous scenario?

Reply:
indeed, it became redundant when I unified the constraintException code :+1: